### PR TITLE
refactor(apig/throttle): refactor throttling policy resource

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Refactor throttling policy resource.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. refactor throttling policy resource.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicy'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicy -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== RUN   TestAccThrottlingPolicy_basic
=== PAUSE TestAccThrottlingPolicy_basic
=== RUN   TestAccThrottlingPolicy_spec
=== PAUSE TestAccThrottlingPolicy_spec
=== CONT  TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicy_spec
=== CONT  TestAccThrottlingPolicy_basic
--- PASS: TestAccThrottlingPolicy_basic (517.85s)
--- PASS: TestAccThrottlingPolicyAssociate_basic (524.93s)
--- PASS: TestAccThrottlingPolicy_spec (528.57s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      528.792s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicyAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicyAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicyAssociate_basic
--- PASS: TestAccThrottlingPolicyAssociate_basic (508.39s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      508.559s
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go [hunk 1]
```diff
@@ -45,43 +47,177 @@ func TestAccThrottlingPolicyAssociate_basic(t *testing.T) {
...
+    name     = "user_age"
+    type     = "NUMBER"
+    location = "PATH"
+    required = true
+    maximum  = 200
+    minimum  = 0
+  }
+  
+  backend_params {
+	type     = "REQUEST"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go]: remove the extra spaces

Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go]: done.

File: docs/resources/apig_throttling_policy.md [hunk 2]
```diff
@@ -60,53 +55,59 @@ resource "huaweicloud_apig_throttling_policy" "test" {
 
 The following arguments are supported:
 
-* `region` - (Optional, String, ForceNew) Specifies the region in which to create the API throttling policy resource.
-  If omitted, the provider-level region will be used. Changing this will create a new API throttling policy resource.
+* `region` - (Optional, String, ForceNew) Specifies the region where the throttling policy is located.
```
Comment (Jason-Zhang9309) [inline docs/resources/apig_throttling_policy.md:58]: extral space at the end of this line

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go [hunk 3]
```diff
@@ -45,43 +49,177 @@ func TestAccThrottlingPolicyAssociate_basic(t *testing.T) {
...
+  vpc_id                = huaweicloud_vpc.test.id
+  subnet_id             = huaweicloud_vpc_subnet.test.id
+  security_group_id     = huaweicloud_networking_secgroup.test.id
+  enterprise_project_id = "0"
+
+  availability_zones = try(slice(data.huaweicloud_availability_zones.test.names, 0, 1), null)
+}
+
+data "huaweicloud_images_image" "test" {
+  name        = "Ubuntu 18.04 server 64bit"
```
Comment (hmbig2) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go]:
```suggestion
  os                  = "Ubuntu"
  most_recent  = true
```

Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_throttling_policy_associate_test.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccThrottlingPolicyAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccThrottlingPolicyAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccThrottlingPolicyAssociate_basic
=== PAUSE TestAccThrottlingPolicyAssociate_basic
=== CONT  TestAccThrottlingPolicyAssociate_basic
--- PASS: TestAccThrottlingPolicyAssociate_basic (553.09s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      553.361s
```

**Pull request merged**